pkg/contexts/credentials/utils.go

```diff
@@ -28,3 +32,63 @@ func CredentialsFor(ctx ContextProvider, obj interface{}, uctx ...UsageContext)
 	}
 	return CredentialsForConsumer(ctx, id)
 }
+
+func GuessConsumerType(ctxp ContextProvider, spec string) string {
+	matchers := ctxp.CredentialsContext().ConsumerIdentityMatchers()
+	lspec := strings.ToLower(spec)
+
+	if matchers.Get(spec) == nil {
+		fix := ""
+		for _, i := range matchers.List() {
+			idx := strings.Index(i.Type, ".")
+			if idx > 0 && i.Type[:idx] == spec {
+				fix = i.Type
+				break
+			}
+		}
+		if fix == "" {
+			for _, i := range matchers.List() {
+				if strings.ToLower(i.Type) == lspec {
+					fix = i.Type
+					break
+				}
+			}
+		}
+		if fix == "" {
+			for _, i := range matchers.List() {
+				idx := strings.Index(i.Type, ".")
+				if idx > 0 && strings.ToLower(i.Type[:idx]) == lspec {
+					fix = i.Type
+					break
+				}
+			}
+		}
+		if fix == "" {
+			min := -1
+			for _, i := range matchers.List() {
+				idx := strings.Index(i.Type, ".")
+				if idx > 0 {
+					d := levenshtein.DistanceForStrings([]rune(lspec), []rune(strings.ToLower(i.Type[:idx])), levenshtein.DefaultOptions)
+					if d < 5 && fix == "" || min > d {
+						fix = i.Type
+						min = d
+					}
+				}
+			}
+		}
+		if fix == "" {
+			min := -1
+			for _, i := range matchers.List() {
+				d := levenshtein.DistanceForStrings([]rune(lspec), []rune(strings.ToLower(i.Type)), levenshtein.DefaultOptions)
+				if d < 5 && fix == "" || min > d {
+					fix = i.Type
+					min = d
+				}
+			}
+		}
+		if fix != "" {
+			return fix
+		}
+	}
+	return spec
+}
```

Comment:
Cool and nice feature, but not related to `check` command... wouldn't a separate PR make sense?